chevah/compat/posix_filesystem.py

```diff
@@ -384,6 +385,32 @@ def openFileForWriting(self, segments, utf8=False):
                     DEFAULT_FILE_MODE)
                 return os.fdopen(fd, 'wb')
 
+    def openFileForUpdating(self, segments, utf8=False):
+        """
+        See `ILocalFilesystem`.
+        """
+        path = self.getRealPathFromSegments(segments)
+        path_encoded = self.getEncodedPath(path)
+
+        self._requireFile(segments)
+        with self._IOToOSError(path), self._impersonateUser():
+            if utf8:
+                # This will also allow reading, but I hope we will
+                # remove this API soon, and we patch the high level API.
+                file = codecs.open(path_encoded, 'r+', 'utf-8')
+
+                def deny_read(size=None):
+                    raise IOError('File not open for reading')
+
+                file.read = deny_read
+                return file
+            else:
+                fd = os.open(
+                    path_encoded,
+                    self.OPEN_WRITE_ONLY,
+                    DEFAULT_FILE_MODE)
+                return os.fdopen(fd, 'wb')
```

Comment:
I notice that we use the following modes: `r+` when supporting UTF8 and `wb` for the rest. The `os.fdopen` API doc sais that the modes are the same as for `open` which eventually `codecs.open` calls.

* ``r+''  Open for reading and writing.  The stream is positioned at the beginning of the file.
* ``w+''  Open for reading and writing.  The file is created if it does not exist, otherwise it is truncated.  The stream is positioned at  the beginning of the file.

I see that the position is set at the beginning on both cases but the second one mentions it will truncate the file. 

I will double check if we have a test for both variants (UTF8/other) and that we check for truncation.